### PR TITLE
refactor(router): api-key routes refactoring

### PR DESCRIPTION
**Pull request by racnan**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates

## Description
- Refactored API keys routes to use `/api_keys/{merchant_id}`.
- Refactored authentication errors.

### Additional Changes

- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
These changes enable extending projects to have merchant based authentication for api routes.

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/services/authentication.rs [hunk 1]
```diff
@@ -87,7 +92,13 @@ impl AuthenticateAndFetch<storage::MerchantAccount, AppState> for MerchantIdAuth
...
             .find_merchant_account_by_merchant_id(self.0.as_ref())
             .await
-            .map_err(|error| error.to_not_found_response(errors::ApiErrorResponse::Unauthorized))
+            .map_err(|e| {
+                if e.current_context().is_db_not_found() {
+                    return errors::ApiErrorResponse::Unauthorized;
+                }
+                errors::ApiErrorResponse::InternalServerError
+            })
+            .into_report()
```
Comment (dracarys18) [inline crates/router/src/services/authentication.rs:101]:
```suggestion
                if e.current_context().is_db_not_found() {
                    e.change_context(errors::ApiErrorResponse::Unauthorized)
                } else {
                    e.change_context(errors::ApiErrorResponse::InternalServerError)
                }
            })
```

File: crates/router/src/services/authentication.rs [hunk 2]
```diff
@@ -107,8 +118,13 @@ impl AuthenticateAndFetch<storage::MerchantAccount, AppState> for PublishableKey
...
             .await
-            .change_context(errors::ApiErrorResponse::Unauthorized)
-            .attach_printable("Merchant not authenticated")
+            .map_err(|e| {
+                if e.current_context().is_db_not_found() {
+                    return errors::ApiErrorResponse::Unauthorized;
+                }
+                errors::ApiErrorResponse::InternalServerError
+            })
+            .into_report()
```
Comment (dracarys18) [inline crates/router/src/services/authentication.rs:127]:
```suggestion
               if e.current_context().is_db_not_found() {
                    e.change_context(errors::ApiErrorResponse::Unauthorized)
                } else {
                    e.change_context(errors::ApiErrorResponse::InternalServerError)
                }
            })
```

**Review by dracarys18 — CHANGES_REQUESTED** [review]

**Comment by racnan** [comment]
@dracarys18 bro resolved comment. Can you check ?

**Review by dracarys18 — DISMISSED** [review]

**Review by dracarys18 — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

File: crates/router/src/routes/api_keys.rs [hunk 3]
```diff
@@ -66,9 +63,9 @@ pub async fn api_key_create(
 pub async fn api_key_retrieve(
     state: web::Data<AppState>,
     req: HttpRequest,
-    path: web::Path<String>,
+    path: web::Path<(String, String)>,
 ) -> impl Responder {
-    let key_id = path.into_inner();
+    let (key_id, _merchant_id) = path.into_inner();
```
Comment (SanchithHegde) [inline crates/router/src/routes/api_keys.rs:68]: Won't the `key_id` and `merchant_id` variables have interchanged values? Same with update and revoke endpoints.

**Pull request merged**